programs/chat/src/instructions/close_chat_permissions_v0.rs

```diff
@@ -0,0 +1,24 @@
+use crate::state::*;
+use anchor_lang::prelude::*;
+
+#[derive(Accounts)]
+pub struct CloseChatPermissionsV0<'info> {
+  /// CHECK: Just get the refunded sol
+  #[account(mut)]
+  pub refund: UncheckedAccount<'info>,
+  #[account(
+    constraint = chat.admin.unwrap() == admin.key()
```

Comment:
Not a big deal, but this instruction isn't callable if the close_chat_v0 instruction is called first. Maybe combine them into one instruction? Otherwise just something to be aware of on frontend.

Reply:
Yeah. I'm going to make sure chat can't be closed before permissions closed. Good catch